docs/api.asciidoc

```diff
@@ -0,0 +1,159 @@
+[[api]]
+== Public API
```

Comment:
Why do you split it into `Client`, `Error` and `Transaction` API when you actually need a `client` instance for all of them?

How about calling it a `Client API` and then only use `Error` and `Transaction` later on?

Reply:
This is one of the things I struggled with when structuring the docs. I though by putting the "Client API" first, the reader will know further down how to get a client.

Reply:
One definitly should find his way around by the explanations you provide. I just think it is somehow confusing to describe it as three seperate APIs. I would only distinguish between the `elasticapm` and the `elasticapm.Client` API. And the split up `elasticapm.Client` into 
- getting an instance
- error capturing
- transaction capturing